src/main/java/net/milkbowl/vault/economy/AbstractEconomy.java

```diff
@@ -16,15 +19,37 @@ public boolean hasAccount(OfflinePlayer player, String worldName) {
 		if (player.getName() == null) return false;
 		return hasAccount(player.getName(), worldName);
 	}
+	
+	@Override
+	public boolean hasAccount(UUID uuid) {
+		if (uuid == null) return false;
+		return hasAccount(Bukkit.getOfflinePlayer(uuid));
+	}

```

Comment:
Is this meant to be player-only? Same goes with methods beneath this.

Reply:
So the way I remember this class working, is that most economy plugins will end up replacing all of these methods, the economy plugins that will actually run these methods are the ones which are still in included in the Vault source:
![Capture](https://user-images.githubusercontent.com/879756/135104861-aa043d11-7c7a-41a0-9399-7b31253b5270.PNG)

Of these plugins the only one I see being used is EssentialsEconomy, which afaik, has already internalized their Vault implementation in their 2.19 version. The economy plugin I maintain (ico5) was dropped from Vault long-ago. Gringott's is the only other one that I see people wanting to use (but it is not maintained any longer.)

**Before this PR is ever accepted something has to be done to avoid calling Bukkit.getOfflinePlayer()**, as this should never be used for cost/time/lag reasons. One way to go about this might be to drop all Economy plugins from Vault.

Reply:
Yah for this PR to get accepted I basically would need to remove all the implementations from Vault. At least we are on the same page as far as that's concerned.

Reply:
I think its about time. Further down you can see my comment where I went and dug up when those plugins were last updated. Did you want some help/a pull request made up for Vault?

Further, when they are removed that means the entire AbstractEconomy class can be removed from VaultAPI, correct?